Examples/SwiftUIExample/SwiftUIExample/CustomPlugins/DynamicUserAgentPlugin.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  DynamicUserAgentPlugin.swift
+//  SwiftUIExampleApp
+//
+//  Created by Satheesh Kannan on 04/09/25.
+//
+
+import Foundation
+import RudderStackAnalytics
+
+#if canImport(WebKit)
+import WebKit
+#endif
+
+// MARK: - DynamicUserAgentPlugin
+/**
+ A plugin that dynamically adds User Agent information to the event payload.
+
+ ## Usage:
+ ```swift
+ // Create and add the plugin
+ let userAgentPlugin = DynamicUserAgentPlugin()
+ analytics.add(plugin: userAgentPlugin)
+ ```
+ */
+final class DynamicUserAgentPlugin: Plugin {
+    var pluginType: PluginType = .preProcess
+    var analytics: Analytics?
+    var userAgent: String?
+    
+    init() {
+        // Read user agent on initialization on main thread
+        Task { @MainActor [weak self] in
+            guard let self else {
+                LoggerAnalytics.debug(log: "Failed to read user agent")
+                return
+            }
+            self.userAgent = await self.readUserAgent()
```

Comment:
Let's document the caveat to move this async operation outside the plugin (before SDK initialisation) for robustness.